core/src/main/scala/org/http4s/MessageFailure.scala

```diff
@@ -92,12 +92,17 @@ sealed abstract class UnsupportedMediaTypeFailure(expected: Set[MediaRange]) ext
 final case class MediaTypeMissing(expected: Set[MediaRange])
   extends UnsupportedMediaTypeFailure(expected)
 {
-  val message = s"Decoder is unable to decode a Message without a MediaType. Expected media ranges: $expected"
+  val expectedMsg = expected.map(_.renderString).mkString(", ")
+  val message = s"No media type specified in Content-Type header. Expected one of the following media ranges: $expectedMsg"
 }
 
 /** Indicates that no [[EntityDecoder]] matches the [[MediaType]] of the [[Message]] being decoded */
 final case class MediaTypeMismatch(messageType: MediaType, expected: Set[MediaRange])
   extends UnsupportedMediaTypeFailure(expected)
 {
-  def message = s"$messageType is not a supported media type. Decoder accepts one of the following media ranges: $expected"
+  def message = {
+    val actual = messageType.renderString
+    val expectedMsg = expected.map(_.renderString).mkString(", ")
+    s"$actual is not a supported media type. Expected one of the following media ranges: $expectedMsg"
```

Comment:
In a reflected attack, you trick a user into submitting a request containing a malicious script, which the server then echoes back, causing the user to execute it.  The simplest example is a link with a malicious query string.  XSS attacks via HTTP header can and do exist.

In this case, it seems that our parsed header model discards the malformed header:

```scala
scala> val req = Request().putHeaders(Header.Raw("Content-Type".ci, """alert("pwn3d");"""));
req: org.http4s.Request = Request(method=GET, uri=/, headers=Headers(Content-Type: alert("pwn3d");)

scala> implicitly[EntityDecoder[String]].decode(req, true).run.run
<console>:18: warning: method run in class Task is deprecated: use unsafePerformSync
              implicitly[EntityDecoder[String]].decode(req, true).run.run
                                                                      ^
res12: scalaz.\/[org.http4s.DecodeFailure,String] = -\/(org.http4s.MediaTypeMissing: Decoder is unable to decode a Message without a MediaType. Expected media ranges: Set(MediaRange(text/*)))
```

I *think* we're safe in this case, but I still tend to err on the side of being conservative on reflected values.